docs/deployment/universal-profile.md

```diff
@@ -159,31 +200,242 @@ await lspFactory.LSP3UniversalProfile.deploy({
 });
 ```
 
-## Uploading LSP3 metadata to IPFS
+### Uploading LSP3 metadata to IPFS
 
-If you wish to upload your LSP3 metadata before deploying you can do so using the static `uploadProfileData` method. This uses the same `lsp3Profile` object schema defined above when deploying a Universal Profile:
+You can upload your LSP3 metadata before deploying a Universal Profile by using the `uploadProfileData` method. This uses the same `lsp3Profile` object schema defined [above](./universal-profile#adding-lsp3-metadata) when deploying a Universal Profile.
+
+Returns an object containing the IPFS upload location of your metadata, and your lsp3metdata as a javascript object.
 
 ```javascript
-const uploadResult = await lspFactory.LSP3UniversalProfile.uploadProfileData({
-  ...myUniversalProfileData,
+myLSPFactory.LSP3UniversalProfile.uploadProfileData(
+  lsp3Profile,
+  uploadOptions?
+);
+```
+
+To upload using a custom IPFS gateway pass the `uploadOptions` object. This is the same `uploadOptions` object used when deploying a Universal Profile. Read more [here](./universal-profile#uploadoptions).
+
+`uploadProfileData` is available as both a static or non-static method so can be called without instantiating an LSPFactory object.
+
+```javascript title="calling uploadProfileData on LSPFactory instance"
+const uploadResult = await myLSPFactory.LSP3UniversalProfile.uploadProfileData(
+  myUniversalProfileData
+);
+
+const myUniversalProfileMetadataJSON = uploadResult.profile;
+/**
+{
+  LSP3Profile: {
+    name: "My Universal Profile",
+    description: "My Cool Universal Profile"
+  }
+}
+*/
+
+const myUniversalProfileMetadataUrl = uploadResult.url;
+// 'https://ipfs.lukso.network/ipfs/QmPzUfdKhY6vfcLNDnitwKanpm5GqjYSmw9todNVmi4bqy'
+```
+
+```javascript title="static uploadProfileData call"
+const uploadResult = await LSP3UniversalProfile.uploadProfileData(
```

Comment:
Might be good to add the import statement at the top, to show you are importing only this static method.